### ISSUE TITLE
Feature Request: Include PyTorch-Geometric in pytorch-fortran

### ISSUE THREAD

**Issue by sebk26** [alexeedm/pytorch-fortran#10]
# Feature Request
Hi @alexeedm,

for my studies I included the pytorch package "PyTorch-Geometric" [1] in the torchfortran bindings to be able to open and do inferencing on models including torch-geometric modules.
This works well and was not too complicated. However, I see that this would create more dependencies for this repository, which is probably not desired.

_My question now would be, if there is an elegant way that my solution could be included somehow so that people with the same intentions don't have to go the whole way or at least see this sample implementation?_

[1] https://github.com/pyg-team/pytorch_geometric

---
## How I included the package
* build cpp backend of TorchSparse [2] and TorchScatter [3] (subpackages of PyG) in `./pyg/`
* include both in `CMAKE_PREFIX_PATH`

[2] https://github.com/rusty1s/pytorch_sparse
[3] https://github.com/rusty1s/pytorch_scatter

#### In `make_gnu.sh`
```bash
PYG="./pyg"
CMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH:"$PYG/pytorch_scatter/install/share/cmake:$PYG/pytorch_sparse/install/share/cmake
```

#### In `src/proxy_lib/torch_proxy.cpp`
```cpp
#include <torchscatter/scatter.h>
#include <torchsparse/sparse.h>
```
#### `src/proxy_lib/CMakeLists.txt`
```cmake
find_package(TorchScatter REQUIRED)
find_package(TorchSparse REQUIRED)
find_package(Python3 COMPONENTS Development)
```
```cmake
target_link_libraries(pytorch_proxy PRIVATE TorchScatter::TorchScatter)
target_link_libraries(pytorch_proxy PRIVATE TorchSparse::TorchSparse)
```

**Comment by alexeedm** [comment]
Hey @sebk26 , I'm glad you found this package useful!

Let me first understand why did you have to include two extra `#include`s in the `torch_proxy.cpp`
Does the code in these headers somehow change the regular API? At a first glance there's just function declarations in `sparse.h` for example. And you never call these functions directly, right? They are supposed to be called by pytorch interpreter?

I mean, if the setup only requires the extra library functions, then we should be able to dlopen the `*.so` files if needed

**Comment by sebk26** [comment]
Yes, exactly. They do not change the API and I don't call them directly. 
My model is build with PyG and the python interpreter did not know what to do with a e.g. ``torch_scatter()` module in the jitted model when I tried to open it in Fortran.

I've never done dynamic linking in cpp (this is what you suggested right?) so I can't propose any modifications that fullfil your suggestion.
